src/CADL.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -189,6 +193,41 @@ function createModel(program: Program): any {
     }
 }
 
+function processServiceAuthentication(
+    authentication: ServiceAuthentication
+): InputAuth {
+    const auth = {} as InputAuth;
+    for (const option of authentication.options) {
+        for (const schema of option.schemes) {
+            switch (schema.type) {
+                case "apiKey":
+                    auth.ApiKey = schema.name;
+                    break;
+                case "oauth2":
+                    let scopes: string[] = [];
+                    for (const flow of schema.flows) {
+                        switch (flow.type) {
```

Comment:
I'd say we should just check if `flow` has `scopes`, `flow.type` is irrelevant.

Reply:
When the schema is 'oauth2' type, it will contain flows and each flow will have type to identify its related definition. if it is 'clientCredentials', it will contain scopes.

Reply:
All currently defined flows have scopes:
- https://github.com/microsoft/cadl/blob/main/packages/rest/lib/auth.cadl#L96
- https://github.com/microsoft/cadl/blob/main/packages/rest/lib/auth.cadl#L106
- https://github.com/microsoft/cadl/blob/main/packages/rest/lib/auth.cadl#L116
- https://github.com/microsoft/cadl/blob/main/packages/rest/lib/auth.cadl#L126